### PR TITLE
Add a landing page for the Usage section

### PR DESCRIPTION
**Pull request by tmjd**
One alternative to this would be to do the same kind of organization in the links on the left.

I considered adding some description sections to some of the bullets but I think the information at the links should provide that information.

### CODE REVIEW

**Comment by tmjd** [comment]
Addresses landing page mentioned in #30

File: master/usage/index.md [hunk 1]
```diff
@@ -1,5 +1,67 @@
...
+- [Calico policy controller]({{site.baseurl}}/{{page.version}}/reference/policy-controller/configuration)
+
+- Orchestrator plugin
+  - CNI
+    - [Calico cni container]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/installation/hosted/)
+  - Libnetwork
+    - [Calico node container]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
+  - [OpenStack ML2]({{site.baseurl}}/{{page.version}}/usage/openstack/configuration#ml2-ml2_confini)
+
+- calicoctl binary
```
Comment (caseydavenport) [inline master/usage/index.md]: This can just be called `calicoctl`

File: master/usage/index.md [hunk 2]
```diff
@@ -1,5 +1,67 @@
...
+installation, check out the
+[Getting Started]({{site.baseurl}}/{{page.version}}/getting-started)
+section for the components that are needed for your specific orchestrator.
+Here are the common components with links to relevant sections for configuring
+and operating those components.
+
+- [Calico node container]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
+  - [Felix]({{site.baseurl}}/{{page.version}}/reference/node/reference/felix/configuration)
+  - [BGP]({{site.baseurl}}/{{page.version}}/usage/configuration/bgp)
+  - [Running Calico Node Container as a Service]({{site.baseurl}}/{{page.version}}/usage/configuration/as-service)
```
Comment (caseydavenport) [inline master/usage/index.md]: I think all the places we say "Calico Node container" should really be "calico/node container".

File: master/usage/index.md [hunk 3]
```diff
@@ -1,5 +1,67 @@
...
+  - [Configuring a Node IP Address and Subnet]({{site.baseurl}}/{{page.version}}/usage/configuration/node)
+
+- [Calico policy controller]({{site.baseurl}}/{{page.version}}/reference/policy-controller/configuration)
+
+- Orchestrator plugin
+  - CNI
+    - [Calico cni container]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/installation/hosted/)
```
Comment (caseydavenport) [inline master/usage/index.md]: Same here - this should probably be the `calico/cni` container.

File: master/usage/index.md [hunk 4]
```diff
@@ -1,5 +1,67 @@
...
+  - Libnetwork
+    - [Calico node container]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
+  - [OpenStack ML2]({{site.baseurl}}/{{page.version}}/usage/openstack/configuration#ml2-ml2_confini)
```
Comment (caseydavenport) [inline master/usage/index.md]: Why is this one formatted as a single bullet where the others are nested?

File: master/usage/index.md [hunk 5]
```diff
@@ -1,5 +1,67 @@
...
-title: Using Calico
----
-
+---
+title: Using Calico
+---
+
 This section contains information on using Calico.
+
+## Calico software components
```
Comment (caseydavenport) [inline master/usage/index.md]: I want to think about what type of user is going to be coming to the "Usage" section - I imagine it's someone who has already deployed Calico after going through the "Getting Started" section, maybe has walked through a tutorial or two, and is now looking for next steps on how to perform common tasks.

I'm not sure this introduction captures that - we should think a bit more about the right wording.

File: master/usage/index.md [hunk 6]
```diff
@@ -1,5 +1,67 @@
...
+
+There are several Calico components that may be ran as part of your
+installation, check out the
+[Getting Started]({{site.baseurl}}/{{page.version}}/getting-started)
+section for the components that are needed for your specific orchestrator.
+Here are the common components with links to relevant sections for configuring
+and operating those components.
+
+- [Calico node container]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
```
Comment (caseydavenport) [inline master/usage/index.md]: Is a user going to know _when_ he/she wants to click on one of these?  See my comment above about why they're coming to this page.

This is probably a good compilation of various configuration references, but maybe it needs its own landing page with more detail on each of these bullets about why you might want to click each one.

File: master/usage/index.md [hunk 7]
```diff
@@ -1,5 +1,67 @@
...
+
+
+## Concept areas
+
+There are several concepts that Calico uses, below are
+those concept areas and links to documentation applicable to their
+configuration or operation.
+
+- Datastore
+  - Calico node container
```
Comment (caseydavenport) [inline master/usage/index.md]: Why is calico/node a subsection of Datastore?

File: master/usage/index.md [hunk 8]
```diff
@@ -1,5 +1,67 @@
...
+    - [Node reference]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
+    - [Kubernetes Datastore Driver]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubernetes-datastore/#configuration-details)
+  - CNI
+    - [Calico cni container]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/installation/hosted/)
+  - Libnetwork
+    - [Calico node container]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
+  - calicoctl
+    - [etcdv2]({{site.baseurl}}/{{page.version}}/reference/calicoctl/setup/etcdv2)
+    - [Kubernetes Datastore Driver]({{site.baseurl}}/{{page.version}}/reference/calicoctl/setup/kubernetes)
+- IP Pools
```
Comment (caseydavenport) [inline master/usage/index.md]: Is this section the right way around?  I feel like the main bullet points should be conceptually swapped around - users aren't necessarily going to come looking for "IP Pools", but they might come looking for "How do I configure outbound connectivity?".

**Comment by tmjd** [comment]
@caseydavenport PTAL, I reworked it to be less grouping and bullet points to be more sections with a little text on what the reader may be looking to do.

File: master/usage/index.md [hunk 9]
```diff
@@ -1,5 +1,101 @@
...
-This section contains information on using Calico.
+---
+title: Using Calico
+---
+
+This section assumes you have some familiarity with Calico and have at
+least read through
+[Getting Started]({{site.baseurl}}/{{page.version}}/getting-started)
+for an orchestrator.  The information here will hopefully provide direction
+on achieving some common tasks and possibly some no so common tasks.
```
Comment (ozdanborne) [inline master/usage/index.md]: typo: _not_ so common :wink:

File: master/usage/index.md [hunk 10]
```diff
@@ -1,5 +1,101 @@
...
+  to set or change the IP Address or method used to select it.
+- Some Calico installation options may handle installing and running
+  calico/node but some deployments will want to do this manually, the page
+  [Running Calico Node Container as a Service]({{site.baseurl}}/{{page.version}}/usage/configuration/as-service)
+  helps in those cases.
+- When running Calico in a Kubernetes cluster is is possible to use
+  the Kubernetes datastore, see
+  [Configuring to use the Kubernetes Datastore Driver]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubernetes-datastore/#configuration-details)
+  for configuring calico/node in that situation.
+- For additional configuration refer to the
```
Comment (ozdanborne) [inline master/usage/index.md]: How bout: For a full reference of all flags processed / used / read (i'm not sure which is best), refer to the [calico/node configuration reference]

**Comment by robbrockbank** [comment]
Hi @tmjd,    

This guide feels a little out of place to me.  It’s the “Usage” overview page but is linking to usage, getting started and reference sections.  

I’d prefer the overview to be a brief summary of what this set of guides is for and perhaps have links into those guides with a single sentence description. 

I worry that there is too much content to make this readable.  It’s useful information, but I don’t think it belongs in an overview.  It contains quite a lot of its own content where I feel that if we need additional content it should be in specific guides for that.  For example, if we want an document that says how to launch and configure calico/node then we should have that as a separate doc and put the laucnh and configure docs as sub-sections?

Worth getting @ozdanborne view of this too.

**Comment by tmjd** [comment]
I missed the mark here. I'm closing this PR

**Pull request closed without merging**